### PR TITLE
sanitize createBlob request metadata in AzStorageBlob

### PR DESCRIPTION
**Pull request by berndverst**
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

# Description

Sanitize blob creation request metadata in Azure Storage Blob Binding

### CODE REVIEW

File: bindings/azure/blobstorage/blobstorage.go [hunk 1]
```diff
@@ -462,3 +463,30 @@ func (a *AzureBlobStorage) isValidDeleteSnapshotsOptionType(accessType azblob.De
...
+			delete(metadata, oldkey)
+			metadata[key] = val
+		}
+
+		// Remove all non-ascii characters and replace all colons with underscores
+		metadata[key] = strings.Map(func(r rune) rune {
+			if r > unicode.MaxASCII {
+				return -1
+			}
+			return r
```
Comment (shubham1172) [inline bindings/azure/blobstorage/blobstorage.go:487]: "Replace colon with underscore" needs to be added.

Comment (berndverst) [reply bindings/azure/blobstorage/blobstorage.go:487]: No the comment needs to be updated. I used to have this before, but it isn't needed :)

Comment (shubham1172) [reply bindings/azure/blobstorage/blobstorage.go:487]: Sure, let's update the `special_value` test as well then.

Comment (berndverst) [reply bindings/azure/blobstorage/blobstorage.go:488]: Done.

**Review by shubham1172 — DISMISSED** [review]

**Review by mukundansundar — APPROVED** [review]
Lgtm

**Pull request merged**